hooks/withSession.js

```diff
@@ -0,0 +1,11 @@
+import { useSession } from "next-auth/react";
+
+export const withSession = (Component) => (props) => {
+  const session = useSession();
+
+  if (Component.prototype.render) {
+    return <Component session={session} {...props} />;
+  }
+
+  throw new Error("Use `useSession` directly");
```

Comment:
Log the session errors in winston.

Reply:
Winston [cannot log in the front end](https://github.com/winstonjs/winston-daily-rotate-file/issues/339#issuecomment-1182901700), so I have refrained from doing it as of now. Moreover, `Error("Use useSession directly")` is not about the session itself, hence I decided not to include it.